### PR TITLE
Prefs[TASK][28590] Create UI (Hotel Tab)

### PR DESCRIPTION
**Pull request by hieuna-441907**
## Related Tickets
- [#Task 28590: Create UI](https://edu-redmine.sun-asterisk.vn/issues/28590)

## WHAT
- Update UI Hotel Tab
## HOW
- Update UI Hotel Tab
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (HuyTQ)
--- | --- | --- | --- | --- | ---
General |Code should be self-explanatory. | Get a feel of story reading, while going through the code. Use appropriate name for variables, functions and classes. If you are taking more time to understand the code, then either code needs refactoring or at least comments have to be written to make it clear. | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
General |Is this function or class too big? If yes, is the function or class having too many responsibilities? | | NO |<li>- [x] no</li>|<li>- [x] no</li>  
General | Does the code follow DRY?: Do not Repeat Yourself | Is there any redundant or duplicate code? DRY (Do not Repeat Yourself) principle: The same code should not be repeated more than twice. Consider reusable services, functions and components. Consider generic functions and classes. | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
General |Exception handling and cleanup (dispose) resources | | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
General | Does the code build and execute correctly? | | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
Conventions | Does the source code have end-line errors? | | NO |<li>- [x] no</li>|<li>- [x] no</li>  
Conventions | No hard coding, use constants/configuration values | | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sunasterisk/travelapp/data/models/Destination.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+//                )
+//                add(
+//                    Destination(
+//                        "Madrid, Spain",
+//                        "https://images.lucasfox.com/location/4x3_1600w/B427B4AFB9.jpg",
+//                        11
+//                    )
+//                )
+//            }
+    }
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/travelapp/data/models/Destination.kt]: không để comment

File: app/src/main/java/com/sunasterisk/travelapp/data/models/Destination.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    val title: String,
+    val imageThumbLink: String,
+    val propertiesNumber: Int
+) {
+    constructor(title: String, imageThumbLink: String, propertiesNumber: Int) : this(
+        "",
+        title,
+        imageThumbLink,
+        propertiesNumber
+    )
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/travelapp/data/models/Destination.kt]: điền default value

File: app/src/main/java/com/sunasterisk/travelapp/ui/adapter/HotelDestinationViewHoler.kt [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+                onItemClick(it, adapterPosition)
+            }
+        }
+    }
+
+    fun bindData(destination: Destination) {
+        itemData = destination
+        itemView.run {
+            textViewDestinationNameItem.text = destination.title
+            textViewPropertiesItem.text = "${destination.propertiesNumber} available properties"
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/travelapp/ui/adapter/HotelDestinationViewHoler.kt]: khai báo trong strings.xml

File: app/src/main/java/com/sunasterisk/travelapp/ui/home/hotel/HotelTabFragment.kt [hunk 4]
```diff
@@ -1,15 +1,51 @@
...
+class HotelTabFragment private constructor() : BaseFragment() {
 
     override val layoutResource get() = R.layout.fragment_hotel_tab
 
-    override fun initComponents() = Unit
+    override fun initComponents() {
+
+        val linearLayoutManager = LinearLayoutManager(context, RecyclerView.VERTICAL, false).let {
+            recyclerViewDestination.layoutManager = it
+        }
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/travelapp/ui/home/hotel/HotelTabFragment.kt]: sao không khai báo trong xml nhỉ

File: app/src/main/java/com/sunasterisk/travelapp/ui/adapter/HotelDestinationViewHoler.kt (hunk 3 above)
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/travelapp/ui/adapter/HotelDestinationViewHoler.kt]: khai báo trong strings.xml

Comment (hieuna-441907) [reply app/src/main/java/com/sunasterisk/travelapp/ui/adapter/HotelDestinationViewHoler.kt]: ok ông

File: app/src/main/java/com/sunasterisk/travelapp/ui/home/hotel/HotelTabFragment.kt (hunk 4 above)
Comment (hieuna-441907) [reply app/src/main/java/com/sunasterisk/travelapp/ui/home/hotel/HotelTabFragment.kt]: ok, tại t quen như vậy

File: app/src/main/java/com/sunasterisk/travelapp/data/models/Destination.kt (hunk 2 above)
Comment (hieuna-441907) [reply app/src/main/java/com/sunasterisk/travelapp/data/models/Destination.kt]: ok

File: app/src/main/java/com/sunasterisk/travelapp/data/models/Destination.kt (hunk 1 above)
Comment (hieuna-441907) [reply app/src/main/java/com/sunasterisk/travelapp/data/models/Destination.kt]: ok

File: app/src/main/java/com/sunasterisk/travelapp/ui/adapter/HotelDestinationViewHoler.kt [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            }
+        }
+    }
+
+    fun bindData(location: Location) {
+        itemData = location
+        itemView.run {
+            textViewDestinationNameItem.text = location.name
+            textViewPropertiesItem.text =
+                "${location.propertiesNumber} ${resources.getString(R.string.title_properties)}"
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/travelapp/ui/adapter/HotelDestinationViewHoler.kt]: Đặt resource dạng 
`"%d properldksjflksdf"`
klhi đó sẽ gọi bằng cách 
`resources.getString(R.string.title_properties, number)`

File: app/src/main/res/layout/include_hotel_tab_front.xml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        android:textSize="@dimen/sp_14"
+        android:textColor="@color/color_grey_40"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <View
+        android:id="@+id/viewDivider"
+        android:layout_width="0dp"
+        android:layout_height="1dp"
```
Comment (huytq-2422) [inline app/src/main/res/layout/include_hotel_tab_front.xml]: hard code

File: app/src/main/res/layout/include_hotel_tab_front.xml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/recyclerViewDestination"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="@dimen/dp_16"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/viewDivider"
+        app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"/>
```
Comment (huytq-2422) [inline app/src/main/res/layout/include_hotel_tab_front.xml]: thêm orientation 
listitem 
itemcount 
vào để design trực quan hơn

File: app/src/main/res/values/dimens.xml [hunk 8]
```diff
@@ -1,6 +1,14 @@
...
+    <dimen name="dp_8">8dp</dimen>
+    <dimen name="dp_12">12dp</dimen>
     <dimen name="dp_15">15dp</dimen>
+    <dimen name="dp_16">16dp</dimen>
+    <dimen name="dp_54">54dp</dimen>
     <dimen name="dp_64">64dp</dimen>
+    <dimen name="dp_72">72dp</dimen>
     <dimen name="dp_172">172dp</dimen>
+    <dimen name="sp_14">14sp</dimen>
+    <dimen name="dp_0">0dp</dimen>
```
Comment (huytq-2422) [inline app/src/main/res/values/dimens.xml]: remove

Comment (hieuna-441907) [reply app/src/main/res/values/dimens.xml]: có thuộc tính cardElevation t muốn để = 0dp thì làm tn?

File: app/src/main/res/layout/include_hotel_tab_front.xml (hunk 7 above)
Comment (hieuna-441907) [reply app/src/main/res/layout/include_hotel_tab_front.xml]: ok

File: app/src/main/res/layout/include_hotel_tab_front.xml (hunk 6 above)
Comment (hieuna-441907) [reply app/src/main/res/layout/include_hotel_tab_front.xml]: ok, t để layout_height ra string resource

File: app/src/main/res/values/dimens.xml (hunk 8 above)
Comment (huytq-2422) [reply app/src/main/res/values/dimens.xml]: thì là 0dp thôi

Comment (hieuna-441907) [reply app/src/main/res/values/dimens.xml]: vậy ok

File: app/src/main/java/com/sunasterisk/travelapp/ui/adapter/HotelDestinationViewHoler.kt (hunk 5 above)
Comment (hieuna-441907) [reply app/src/main/java/com/sunasterisk/travelapp/ui/adapter/HotelDestinationViewHoler.kt]: ok, t đã sửa

**Review by tuannm-1276 — APPROVED** [review]

File: app/src/main/java/com/sunasterisk/travelapp/ui/adapter/HotelDestinationViewHoler.kt [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+            }
+        }
+    }
+
+    fun bindData(location: Location) {
+        itemData = location
+        itemView.run {
+            textViewDestinationNameItem.text = location.name
+            textViewPropertiesItem.text =
+                "${resources.getString(R.string.title_properties, location.propertiesNumber)}"
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/travelapp/ui/adapter/HotelDestinationViewHoler.kt]:
```suggestion
            textViewPropertiesItem.text =
                resources.getString(R.string.title_properties, location.propertiesNumber)
```

File: app/src/main/java/com/sunasterisk/travelapp/ui/adapter/HotelDestinationsAdapter.kt [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+    var onItemClick: (Location, Int) -> Unit = { _, _ -> }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HotelDestinationViewHolder {
+        val itemView =
+            LayoutInflater.from(parent.context).inflate(R.layout.item_destination, parent, false)
+        return HotelDestinationViewHolder(itemView, onItemClick)
+    }
+
+    override fun getItemCount() = 12
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/travelapp/ui/adapter/HotelDestinationsAdapter.kt]: sao lại fix cứng 12 ở đây nhỉ

File: app/src/main/java/com/sunasterisk/travelapp/ui/adapter/HotelDestinationsAdapter.kt [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.sunasterisk.travelapp.R
+import com.sunasterisk.travelapp.data.models.Location
+
+class HotelDestinationsAdapter(private val destinations: List<Location>) :
+    RecyclerView.Adapter<HotelDestinationViewHolder>() {
+
+    var onItemClick: (Location, Int) -> Unit = { _, _ -> }
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/travelapp/ui/adapter/HotelDestinationsAdapter.kt]: Tôi tưởng ông có base recycler view adapter nhỉ

Comment (hieuna-441907) [reply app/src/main/java/com/sunasterisk/travelapp/ui/adapter/HotelDestinationsAdapter.kt]: ok, để t tạo thêm

File: app/src/main/java/com/sunasterisk/travelapp/ui/adapter/HotelDestinationsAdapter.kt (hunk 10 above)
Comment (hieuna-441907) [reply app/src/main/java/com/sunasterisk/travelapp/ui/adapter/HotelDestinationsAdapter.kt]: úi, t quên mất

File: app/src/main/java/com/sunasterisk/travelapp/ui/adapter/HomeTabsAdapter.kt [hunk 12]
```diff
@@ -19,4 +19,6 @@ class HomeTabsAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {
         fragments.add(fragment)
         fragmentTitles.add(title)
     }
+
+
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/travelapp/ui/adapter/HomeTabsAdapter.kt:23]: sau khi push xong đọc 1 lượt tránh mấy cái sót này nhá

Comment (hieuna-441907) [reply app/src/main/java/com/sunasterisk/travelapp/ui/adapter/HomeTabsAdapter.kt:23]: ok ông

**Pull request merged**